### PR TITLE
Versioning proposition

### PR DESCRIPTION
**Pull request by radepal**

### CODE REVIEW

File: docs/versioning.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+## Tagging
+
+Git tags MUST follow MAJOR.MINOR.PATCH convention eg. 0.0.1, 0.1.0, 1.0.0  on master branch.
+Git tags on non master branches (features to master) should add -rc -alfa -beta if needed for major changes.
+Git tags for patches for older releases could be done on stable/fix branch eg. 1.2.3
+
+## Developing 
+
+New changes are developed on branch from master. Testing changes from dependent modules/repos are done on separate branch. To upgrade dependency first that repos should be tagged and than upgraded in dependant repo.
```
Comment (mzawadzk) [inline docs/versioning.md]: s/than/then

File: docs/versioning.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+## Deployment
+
+Version of Kentrkios project is based on this repo version. All other dependencies as an requirement in REQUIREMENTS file as repo and version.
```
Comment (mzawadzk) [inline docs/versioning.md]: s/Kentrkios/Kentrikos
I suggest:
This repository (aws-bootstrap) is considered top-level repository for Kentrikos project and its version is bound to required versions of all other dependent repositories as described in REQUIREMENTS file.

Comment (mzawadzk) [inline docs/versioning.md]: Also I suggest adding sub-sections:
### How to deploy from latest stable version
### How to deploy from latest development version

File: docs/versioning.md (hunk 1 above)
Comment (mzawadzk) [inline docs/versioning.md]: I suggest:
New changes should be developed and tested using features branches (forked from master). Dependencies should be upgraded by tagging repositories top to bottom.
e.g.: [please add an example]

File: docs/versioning.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+### Branching
+
+* Master branch is for stable, tested changes.
+* Features are developing on separate branches that ara checkout from master
+
+## Deployment
+
+This repository (aws-bootstrap) is considered top-level repository for Kentrikos project and its version is bound to required versions of all other dependent repositories as described in REQUIREMENTS file as repo:version per line
+eg.
+Giving provided below  structure of dependency, change in terraforma-ws-kops module must be updated and tagged from bottom to top level repo.
```
Comment (mzawadzk) [inline docs/versioning.md]: typo: terraform-aws-kops

File: docs/versioning.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+  |
+   \template-environment-configuration
+    |
+     \terraform-aws-kops
+```
+
+### How to deploy from latest stable version
+
+1. Take last version from master branch of this repository (aws-bootstrap) 
+1. Follow instruction on "Environment configuration" section.
```
Comment (mzawadzk) [inline docs/versioning.md]: instructions

File: docs/versioning.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+  //...
+}
+```
+
+## Environment configuration
+
+To update environment configuration we should base on desire version of Kentrikos project.
+
+1. From aws-bootsrap repo we chouse proper version
```
Comment (mzawadzk) [inline docs/versioning.md]: choose

File: docs/versioning.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+1. From REQUIREMENTS we take depended version of template-environment-configuration repo
```
Comment (mzawadzk) [inline docs/versioning.md]: s/depended/required ?

File: docs/versioning.md [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+module "jenkins" {
+  source = "github.com/kentrikos/terraform-aws-bootstrap-jenkins?ref=0.2.0"
+
+  //...
+}
+```
+
+## Environment configuration
+
+To update environment configuration we should base on desire version of Kentrikos project.
```
Comment (mietlinski-nc) [inline docs/versioning.md]:
```suggestion
To update environment configuration we should base on desired version of Kentrikos project.
```

File: docs/versioning.md (hunk 3 above)
Comment (mietlinski-nc) [inline docs/versioning.md]:
```suggestion
Giving provided below structure of dependency, change in terraform-aws-kops module must be updated and tagged from bottom to top level repository. 
```

File: docs/versioning.md (hunk 5 above)
Comment (mietlinski-nc) [inline docs/versioning.md]:
```suggestion
1. From aws-bootsrap repo we choose proper version
```

File: docs/versioning.md [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+
+1. From aws-bootsrap repo we chouse proper version
+1. From REQUIREMENTS we take depended version of template-environment-configuration repo
+1. Check changes from our last version of template-environment-configuration and selected.
```
Comment (mietlinski-nc) [inline docs/versioning.md]:
```suggestion
1. Check changes from last version of template-environment-configuration and selected.
```
mark selected ??

**Pull request merged**